packages/core/snapshot/manager.go

```diff
@@ -0,0 +1,175 @@
+package snapshot
+
+import (
+	"errors"
+	"sync"
+
+	"github.com/iotaledger/goshimmer/packages/core/epoch"
+	"github.com/iotaledger/goshimmer/packages/core/ledger"
+	"github.com/iotaledger/goshimmer/packages/core/notarization"
+	"github.com/iotaledger/goshimmer/packages/core/tangleold"
+	"github.com/iotaledger/goshimmer/packages/node/database"
+	"github.com/iotaledger/hive.go/core/generics/event"
+	"github.com/iotaledger/hive.go/core/kvstore"
+)
+
+const (
+	prefixSolidEntryPoint byte = iota
+)
+
+// Manager is the snapshot manager.
+type Manager struct {
+	tangle          *tangleold.Tangle
+	notarizationMgr *notarization.Manager
+
+	baseStore kvstore.KVStore
```

Comment:
I don't think we should have a persistent storage for solid entrypoints: they are volatile and limited in nature, until we commit to an epoch. Let's use a ShrinkingMap instead.